<?xml version="1.0"?>

<launch>

  <arg name="use_map_topic"  default="true"/> <!--  Jan 5th 2018 -->
  <arg name="scan_topic"     default="scan"/> <!--  Jan 5th 2018 -->
  <arg name="initial_pose_x" default="0.0"/> <!--  Jan 5th 2018 -->
  <arg name="initial_pose_y" default="0.0"/> <!--  Jan 5th 2018 -->
  <arg name="initial_pose_a" default="0.0"/> <!--  Jan 5th 2018 --> 
  <arg name="global_frame_id" default="map" /> <!--  Jan 5th 2018 --> 
  <arg name="tf_prefix"   default="tf_p" /> <!--  Jan 5th 2018 -->

  <arg name="robot" />

  <node pkg="collvoid_amcl" type="collvoid_amcl" name="amcl"> <!-- ns="$(arg robot)"> -->
    <!-- Publish scans from best pose at a max of 10 Hz -->
    <param name="odom_model_type" value="diff"/> <!-- -->
    <param name="transform_tolerance" value="0.2" /> <!-- -->
    <param name="gui_publish_rate" value="10.0"/> <!-- -->
    <param name="laser_max_beams" value="30"/><!-- -->
    <param name="min_particles" value="100"/><!-- -->
    <param name="max_particles" value="500"/><!-- -->
    <param name="kld_err" value="0.05"/><!-- -->
    <param name="kld_z" value="0.5"/><!-- -->
    <param name="odom_alpha1" value="0.2"/><!-- -->
    <param name="odom_alpha2" value="0.2"/><!-- -->
    <!-- translation std dev, m -->
    <param name="odom_alpha3" value="0.2"/><!-- -->
    <param name="odom_alpha4" value="0.2"/><!-- -->
    <param name="laser_z_hit" value="0.5"/><!-- -->
    <param name="laser_z_short" value="0.05"/><!-- -->
    <param name="laser_z_max" value="0.05"/><!-- -->
    <param name="laser_z_rand" value="0.5"/><!-- -->
    <param name="laser_sigma_hit" value="0.2"/><!-- -->
    <param name="laser_lambda_short" value="0.1"/><!-- -->
    <param name="laser_lambda_short" value="0.1"/><!-- -->
    <param name="laser_model_type" value="likelihood_field"/><!-- -->
    <!-- <param name="laser_model_type" value="beam"/> -->
    <param name="laser_likelihood_max_dist" value="2.0"/><!-- -->
    <param name="update_min_d" value="0.1"/><!-- -->
    <param name="update_min_a" value="0.2"/><!-- -->
    <param name="base_frame_id" value="$(arg tf_prefix)/base_link"/>        <!--  Jan 5th 2018 -->
    <param name="odom_frame_id" value="$(arg tf_prefix)/odom"/>             <!--  Jan 5th 2018 -->
    <param name="global_frame_id" value="$(arg global_frame_id)"/>          <!--  Jan 5th 2018 -->

    <param name="initial_pose_x"            value="$(arg initial_pose_x)"/> <!--  Jan 5th 2018 -->
    <param name="initial_pose_y"            value="$(arg initial_pose_y)"/> <!--  Jan 5th 2018 -->
    <param name="initial_pose_a"            value="$(arg initial_pose_a)"/> <!--  Jan 5th 2018 -->
    <remap from="scan"                      to="$(arg scan_topic)"       /> <!--  Jan 5th 2018 -->
    <param name="use_map_topic"             value="$(arg use_map_topic)"/>  <!--  Jan 5th 2018 -->
    <param name="tf_broadcast" value="true"/>                               <!--  Jan 5th 2018 -->
    <param name="tf_prefix" value="$(arg tf_prefix)"/>                      <!--  Jan 5th 2018 -->
    <remap from="map" to="/map" />                                          <!--  Jan 5th 2018 -->
    
    <param name="resample_interval" value="2"/> <!-- -->
    <param name="recovery_alpha_slow" value="0.0"/><!-- -->
    <param name="recovery_alpha_fast" value="0.0"/><!-- -->
    <!-- <remap from="scan" to="base_scan"/> -->                             <!--  Jan 5th 2018 -->
    <remap from="static_map" to="/static_map"/>  <!-- service --> 
  </node>

</launch>
